src/presets/erc20_votes.cairo

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts for Cairo v0.11.0 (presets/erc20_votes.cairo)
+
+/// ERC20 with the ERC20Votes extension.
+#[starknet::contract]
+mod ERC20Votes {
+    use openzeppelin::token::erc20::ERC20Component;
+    use openzeppelin::token::erc20::extensions::ERC20VotesComponent::InternalTrait as ERC20VotesInternalTrait;
+    use openzeppelin::token::erc20::extensions::ERC20VotesComponent;
+    use openzeppelin::utils::cryptography::nonces::NoncesComponent;
+    use openzeppelin::utils::cryptography::snip12::SNIP12Metadata;
+    use openzeppelin::utils::structs::checkpoint::Checkpoint;
+    use starknet::ContractAddress;
+
+    component!(path: ERC20VotesComponent, storage: erc20_votes, event: ERC20VotesEvent);
+    component!(path: ERC20Component, storage: erc20, event: ERC20Event);
+    component!(path: NoncesComponent, storage: nonces, event: NoncesEvent);
+
+    // ERC20Votes
+    #[abi(embed_v0)]
+    impl ERC20VotesComponentImpl =
+        ERC20VotesComponent::ERC20VotesImpl<ContractState>;
+
+    // ERC20Mixin
+    #[abi(embed_v0)]
+    impl ERC20MixinImpl = ERC20Component::ERC20MixinImpl<ContractState>;
+    impl InternalImpl = ERC20Component::InternalImpl<ContractState>;
+
+    // Nonces
+    #[abi(embed_v0)]
+    impl NoncesImpl = NoncesComponent::NoncesImpl<ContractState>;
+
+    #[storage]
+    struct Storage {
+        #[substorage(v0)]
+        erc20_votes: ERC20VotesComponent::Storage,
+        #[substorage(v0)]
+        erc20: ERC20Component::Storage,
+        #[substorage(v0)]
+        nonces: NoncesComponent::Storage
+    }
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {
+        #[flat]
+        ERC20VotesEvent: ERC20VotesComponent::Event,
+        #[flat]
+        ERC20Event: ERC20Component::Event,
+        #[flat]
+        NoncesEvent: NoncesComponent::Event
+    }
+
+    /// Required for hash computation.
+    impl SNIP12MetadataImpl of SNIP12Metadata {
```

Comment:
@andrew-fleming even if we decide to use storage, I would need to update the SNIP12 utilities and add a component for that, and change the documentation. I still lean towards keeping the current implementation and removing the preset, but even if we decide to keep it, I think we should remove it for this PR, and add it again on the future after updating the SNIP12 implementations.

Reply:
Maybe we could write an instructional blog post or guide on creating an ERC20Votes contract. We have the wizard, of course, but maybe a guide could help compliment it? I do agree though that removing the preset is the best approach right now